pm_nq/pm_nq.xml

```diff
@@ -0,0 +1,253 @@
+                    <option value="hyperfine">Hyperfine</option>
+                    <option value="charge">Charge</option>
+                </param>
+                <conditional name="calculator">
+                    <param argument="calculator" type="select" display="radio" label="Calculator" help="Used to calculate the property.">
+                        <option value="castep">CASTEP</option>
+                        <option value="dftb+">DFTB+</option>
+                    </param>
+                    <when value="castep">
+                        <param argument="castep_param" type="data" format="txt" optional="true" label="Castep Parameters File" help="For convenience, this file is copied to all output directories (optional)."/>
+                    </when>
+                    <when value="dftb+">
+                        <expand macro="dftb+"/>
+                    </when>
+                </conditional>
+                <param argument="write_allconf" type="boolean" label="Write All Configurations" help="Whether to write a 'collective' file containing all the displaced positions."/>
+                <param argument="displace_T" type="float" value="0" min="0" label="Displacement Temperature" help="Temperature used to generate displacements."/>
+                <param argument="average_T" type="float" optional="true" min="0" label="Averaging Temperature" help="Temperature used when averaging property (optional). If unset, will use the same value as Averaging Temperature."/>
+                <param argument="random_seed" type="integer" optional="true" label="Random Seed" help="Seed to use for random generation."/>
+            </when>
+            <when value="r">
+                <param name="zipped_input" type="data" format="zip" label="Zipped Input" help="A zipped directory structure containing the output of the generate mode, and the results of running CASTEP or DFTB+ to generate the desired property."/>
+            </when>
+        </conditional>
+        <param type="hidden" name="testing" label="Test mode" value="false"/>
+    </inputs>
+    <outputs>
+        <data label="Displaced Structures" name="displaced_structures" format="zip" from_work_dir="out_zip.zip">
+            <filter>(task['task'] == 'w')</filter>
+        </data>
+        <data label="File tree (testing only)" name="file_tree" format="txt" from_work_dir="tree.txt" hidden="true">
+            <filter>(task['task'] == 'w' and testing == 'true')</filter>
+        </data>
+        <data label="Averaged Property" name="averaged_property" format="txt" from_work_dir="averages.dat">
+            <filter>(task['task'] == 'r')</filter>
+        </data>
+    </outputs>
+    <tests>
+        <!-- Test CASTEP without default values -->
+        <test expect_num_outputs="2">
+            <param name="testing" value="true"/>
+            <param name="structure" value="Si.cell" ftype="cell"/>
```

Comment:
Noting that this isn't a muonated structure - the tool still works, but it's not really a 'model' input. Perhaps we could use a file with a muon as created by pm-muairss?
Though that would presumably involve remaking a bunch of other test data too - don't bother if you don't think it's worth the effort.

Reply:
In the case where nothing is labelled a muon, PyMuonSuite will assume the last atom in the structure is the muon. The same thing happens for the `ethyleneMu`, as the muon is actually labelled `H` but is at the bottom of the file.

Obviously the physical validity goes out of the window since it's actually displacing `Si`, but that's not what we're testing here (just that it generates the correct file structure). I'd leave it as is (especially since SCARF is down atm and so I can't generate any new data anyway).